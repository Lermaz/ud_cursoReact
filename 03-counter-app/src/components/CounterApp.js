import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {

  const [ counter, setCounter ] = useState( value );

  // HandleAdd
  const handleAdd = () => {
    setCounter( counter + 1);
  }
  
  const handleRemove = () => {
    setCounter( counter - 1);
  }

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <Fragment>
      <h1>CounterApp</h1>
      <h2>{ counter }</h2>
      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleReset }>Reset</button>
      <button onClick={ handleRemove }>-1</button>
    </Fragment>
  );
}

CounterApp.propTypes = {
  values: PropTypes.number
}

export default CounterApp;