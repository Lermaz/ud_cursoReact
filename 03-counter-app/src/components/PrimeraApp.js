import { Fragment } from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {


  return (
    <Fragment>
      <h1>{ saludo }</h1>
      <p>{ subtitulo }</p>
    </Fragment>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;