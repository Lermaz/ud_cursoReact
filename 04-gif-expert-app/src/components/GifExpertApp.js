import React, { Fragment, useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategories] = useState( defaultCategories );

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />
      <ol>
        {
          categories.map( category => 
             <GifGrid 
              key={ category }
              category={ category } />
          )
        }
      </ol>
    </Fragment>
  )
}