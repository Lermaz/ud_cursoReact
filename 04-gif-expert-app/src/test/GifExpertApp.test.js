import { shallow } from 'enzyme';
import { GifExpertApp } from '../components/GifExpertApp';


describe('Pruebas en GifExpertApp', () => {
  
  test('Debe mostrarse correctamente', () => {

    const wrapper = shallow( <GifExpertApp /> )
    
    expect( wrapper ).toMatchSnapshot();
  });  

  test('Debe de mostrar una lista de categorias', () => {
    
    const categories = ['Mr. Robot', 'Parks and Recreation'];
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  });
})
