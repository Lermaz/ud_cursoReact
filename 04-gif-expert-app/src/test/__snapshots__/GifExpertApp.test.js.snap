// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Mr. Robot"
      key="Mr. Robot"
    />
    <GifGrid
      category="Parks and Recreation"
      key="Parks and Recreation"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
